sktime/decomposition/koopman.py

```diff
@@ -0,0 +1,575 @@
+        return CovarianceKoopmanModel._cumvar(self.singular_values)
+
+    def score(self, test_model=None, score_method='VAMP2'):
+        """Compute the VAMP score for this model or the cross-validation score between self and a second model.
+
+        Parameters
+        ----------
+        test_model : CovarianceKoopmanModel, optional, default=None
+
+            If `test_model` is not None, this method computes the cross-validation score
+            between self and `test_model`. It is assumed that self was estimated from
+            the "training" data and `test_model` was estimated from the "test" data. The
+            score is computed for one realization of self and `test_model`. Estimation
+            of the average cross-validation score and partitioning of data into test and
+            training part is not performed by this method.
+
+            If `test_model` is None, this method computes the VAMP score for the model
+            contained in self.
+
+        score_method : str, optional, default='VAMP2'
+            Available scores are based on the variational approach
+            for Markov processes :cite:`vampscore-wu2020variational`:
+
+            *  'VAMP1'  Sum of singular values of the half-weighted Koopman matrix :cite:`vampscore-wu2020variational`.
+                        If the model is reversible, this is equal to the sum of
+                        Koopman matrix eigenvalues, also called Rayleigh quotient :cite:`vampscore-wu2020variational`.
+            *  'VAMP2'  Sum of squared singular values of the half-weighted Koopman
+                        matrix :cite:`vampscore-wu2020variational`. If the model is reversible, this is
+                        equal to the kinetic variance :cite:`vampscore-noe2015kinetic`.
+            *  'VAMPE'  Approximation error of the estimated Koopman operator with respect to
+                        the true Koopman operator up to an additive constant :cite:`vampscore-wu2020variational` .
+
+        Returns
+        -------
+        score : float
+            If `test_model` is not None, returns the cross-validation VAMP score between
+            self and `test_model`. Otherwise return the selected VAMP-score of self.
+
+        References
+        ----------
+        .. bibliography:: /references.bib
+            :style: unsrt
+            :filter: docname in docnames
+            :keyprefix: vampscore-
+        """
+        if test_model is None:
+            test_model = self
+        Uk = self.singular_vectors_left[:, 0:self.output_dimension]
+        Vk = self.singular_vectors_right[:, 0:self.output_dimension]
+        res = None
+        if score_method == 'VAMP1' or score_method == 'VAMP2':
+            A = spd_inv_sqrt(Uk.T.dot(test_model.cov_00).dot(Uk), epsilon=self.epsilon)
+            B = Uk.T.dot(test_model.cov_0t).dot(Vk)
+            C = spd_inv_sqrt(Vk.T.dot(test_model.cov_tt).dot(Vk), epsilon=self.epsilon)
+            ABC = mdot(A, B, C)
```

Comment:
i've seen this ABC notation before eg in our paper w/ martin but maybe adding a bit to the docs w/ the equation would be nice - sorry if you did and i missed it

Reply:
i can document this better, it is c&p code